src/frontend/screens/CategoryChooser.js

```diff
@@ -71,20 +71,54 @@ const Item = React.memo(
 const CategoryChooser = ({ navigation }: { navigation: NavigationProp }) => {
   const [{ presets }] = useContext(ConfigContext);
   const [{ value: draftValue }, { updateDraft }] = useDraftObservation();
-
   const presetsList = Array.from(presets.values())
     // Sort presets by sort property and then by name, then filter only point presets
     .sort(presetCompare)
     // Only show presets where the geometry property includes "point"
     .filter(p => p.geometry.includes("point"));
-
-  const handleSelectPreset = (preset: Preset) => {
-    updateDraft({
-      tags: {
-        ...(draftValue || {}).tags,
-        categoryId: preset.id,
-      },
-    });
+  const handleSelectPreset = (selectedPreset: Preset) => {
+    if (draftValue && Object.keys(draftValue.tags).length > 0) {
+      // We're updating an observation
+      // Tags from current preset
+      const currentDraftTags = draftValue.tags;
+      // Tags from previous preset
+      const prevPresetTags =
+        (Object.fromEntries(presets)[currentDraftTags.categoryId] || {}).tags ||
```

Comment:
```suggestion
        (presets.get(currentDraftTags.categoryId) || {}).tags ||
```
My suggestion probably was confusing, I thought this was an object, not a Map. Since it's a Map, we can just get the preset directly.